internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_virtual_machine_instance_resource.go

```diff
@@ -0,0 +1,983 @@
+					},
+
+					"dynamic_memory_min_in_mb": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						ValidateFunc: validation.IntBetween(32, 1048576),
+						AtLeastOneOf: []string{"hardware.0.cpu_count", "hardware.0.dynamic_memory_max_in_mb", "hardware.0.dynamic_memory_min_in_mb", "hardware.0.limit_cpu_for_migration_enabled", "hardware.0.memory_in_mb"},
+					},
+
+					"limit_cpu_for_migration_enabled": {
+						Type:         pluginsdk.TypeBool,
+						Optional:     true,
+						AtLeastOneOf: []string{"hardware.0.cpu_count", "hardware.0.dynamic_memory_max_in_mb", "hardware.0.dynamic_memory_min_in_mb", "hardware.0.limit_cpu_for_migration_enabled", "hardware.0.memory_in_mb"},
+					},
+
+					"memory_in_mb": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						ValidateFunc: validation.IntBetween(32, 1048576),
+						AtLeastOneOf: []string{"hardware.0.cpu_count", "hardware.0.dynamic_memory_max_in_mb", "hardware.0.dynamic_memory_min_in_mb", "hardware.0.limit_cpu_for_migration_enabled", "hardware.0.memory_in_mb"},
+					},
+				},
+			},
+		},
+
+		"network_interface": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: networkValidate.NetworkInterfaceName,
+					},
+
+					"virtual_network_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: virtualnetworks.ValidateVirtualNetworkID,
+					},
+
+					"ipv4_address_type": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringInSlice(virtualmachineinstances.PossibleValuesForAllocationMethod(), false),
+					},
+
+					"ipv6_address_type": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringInSlice(virtualmachineinstances.PossibleValuesForAllocationMethod(), false),
+					},
+
+					"mac_address_type": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringInSlice(virtualmachineinstances.PossibleValuesForAllocationMethod(), false),
+					},
+				},
+			},
+		},
+
+		"operating_system": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"computer_name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validate.SystemCenterVirtualMachineManagerVirtualMachineInstanceComputerName,
+					},
+
+					"admin_password": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						Sensitive:    true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"storage_disk": {
```

Comment:
Yes. There is no required properties for `storage_disk` and `name` is not required. Because the sub properties in `storage_disk` are set in different combinations.

Reply:
isn't `lun` required? Can it be duplicated?

Reply:
Nope. `lun` is not required. For example, when "template_disk_id" is set, then other properties don't need to be set.